src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -477,6 +477,14 @@ protected function forward(Request $request, $catch = false, Response $entry = n
             $request->headers->set('X-Forwarded-For', $request->server->get('REMOTE_ADDR'));
         }
 
+        // modify the Forwarded header if needed
+        $forwarded = $request->headers->get('Forwarded');
+        if ($forwarded) {
+            $request->headers->set('Forwarded', preg_replace('/for=([0-9a-zA-Z.,\\s]*)/', '$0, '.$request->server->get('REMOTE_ADDR'), $forwarded));
```

Comment:
right, but then why have this preg_replace just to concat? if there is no 'for', ths will miss adding the new one, isn't it?